docs/API.md

```diff
@@ -1702,10 +1703,29 @@ The above example is equivalent to the `meta/common-info` example above.
 
 **Not yet implemented**
 
-#### POST *id*/resources/name.stream
+#### GET *id*/meta/resources
 
-Posting to the resources path creates a new version of the given stream
-for the charm with the given id. The request returns the new version.
+Getting from this path will return metadata regarding the resources for the charm with id of `id`.
+
+```go
+type ResourcesResponseP struct {
+    Resources []Resource
+}
+
+type Resource struct {
+    Name        string
+    Type        string
+    Path        string
+    Description string
+    Revision    int
+    Fingerprint string
+    Size        int64
+}
+```
+
+#### POST *id*/resources/name
```

Comment:
Yeah, I had the same thought, but figured this was the way the charmstore folks wanted the workflow structured. I'll propose a unified PUT and see what feedback they have.